Tests/RoslynPluginGeneratorTests/AnalyzerPluginGeneratorTests.cs

```diff
@@ -208,48 +293,123 @@ public void Generate_LicenseAcceptanceNotRequestedIfNoAnalysers()
+            logger.AssertWarningsLogged(2);
             logger.AssertErrorsLogged(0);
         }
 
         [TestMethod]
-        public void Generate_SqaleFileNotSpecified_TemplateFileCreated()
+        public void Generate_LicenseAcceptanceRequested_NoAnalysersInTarget()
         {
+            // If there are:
+            // Required licenses
+            // No analyzers in the targeted package, but analyzers in the dependencies
+            // We should fail due to the absence of analyzers if we are only generating a plugin for the targeted package
+            // We should fail with an error due to licenses if we are generating plugins for the targeted package and dependencies
+
             // Arrange
             string outputDir = TestUtils.CreateTestDirectory(this.TestContext, ".out");
+            string dummyContentFile = TestUtils.CreateTextFile("dummy.txt", outputDir, "non-analyzer content file");
+
+            RemoteRepoBuilder remoteRepoBuilder = new RemoteRepoBuilder(this.TestContext);
+
+            // Multi-level: parent and some but not all dependencies require license acceptance
+            // Parent has no analyzers, but dependencies do
+            IPackage child1 = CreatePackageWithAnalyzer(remoteRepoBuilder, "child1.requiredAccept.id", "2.1", License.Required);
+            IPackage child2 = CreatePackageWithAnalyzer(remoteRepoBuilder, "child2.id", "2.2", License.NotRequired);
+            remoteRepoBuilder.CreatePackage("non-analyzer.parent.requireAccept.id", "1.0", dummyContentFile, License.Required, child1, child2);
 
             TestLogger logger = new TestLogger();
+            AnalyzerPluginGenerator apg = CreateTestSubjectWithFakeRemoteRepo(remoteRepoBuilder, logger);
 
-            RemoteRepoBuilder remoteRepoBuilder = new RemoteRepoBuilder(this.TestContext);
-            CreatePackageInFakeRemoteRepo(remoteRepoBuilder, "dummy.id", "1.1");
+            // 1. a) Only target package. User does not accept, but no analyzers so no license prompt -> fails due to absence of analyzers
+            ProcessedArgs args = CreateArgs("non-analyzer.parent.requireAccept.id", "1.0", "cs", null, false /* accept licenses */ ,
+                false, outputDir);
+            bool result = apg.Generate(args);
+            Assert.IsFalse(result, "Expecting generator to fail");
 
-            NuGetPackageHandler nuGetHandler = new NuGetPackageHandler(remoteRepoBuilder.FakeRemoteRepo, GetLocalNuGetDownloadDir(), logger);
+            logger.AssertSingleWarningExists(String.Format(UIResources.APG_NoAnalyzersFound, "non-analyzer.parent.requireAccept.id"));
+            logger.AssertSingleWarningExists(UIResources.APG_NoAnalyzersInTargetSuggestRecurse);
+            logger.AssertWarningsLogged(2);
+            logger.AssertErrorsLogged(0);
 
-            string expectedTemplateSqaleFilePath = Path.Combine(outputDir, "dummy.id.1.1.sqale.template.xml");
```

Comment:
A lot of this code was refactored out into AssertSqaleFileExistsForPackage(), to avoid having a lot of duplication and to make the test more readable.